include/bluetooth/nrf/host_extensions.h

```diff
@@ -116,6 +118,67 @@ int bt_conn_set_power_control_request_params(struct bt_conn_set_pcr_params *para
  */
 int bt_nrf_host_extension_reduce_initator_aux_channel_priority(bool reduce);
 
+/** @brief Bluetooth LTK
+ *
+ * A Bluetooth LTK in little-endian order. The bytes in this structure
+ * are in the same order as in HCI commands.
+ */
+struct nrf_ltk {
+	uint8_t val[16];
+};
+
+/**
+ * @brief Input the encryption key to be used for connection encryption.
+ *
+ * This API does not itself enable encryption. Instead, the encryption
+ * keys is set to be used when encryption is triggered by other means.
+ * Peripherals should wait for the central to trigger encryption, while
+ * Centrals should follow up with a call to @ref bt_conn_set_security.
+ *
+ * This API may be used in conn_cb.connected() callback. It can also be
+ * called later at any time before encryption is triggered. In the
+ * connected callback, encryption is guaranteed not to have been
+ * triggered yet.
+ *
+ * Important: This API is intended to be used as a mechanism for
+ * application-defined connection encryption when both devices have
+ * agreed on a shared secret. Users of this API must ensure this API is
+ * not used on a connection to device unaware of the particular
+ * proprietary scheme this mechanism is used for.
+ *
+ * This is a transitional API that will be deprecated when a similar
+ * feature lands in standard Bluetooth.
+ *
+ * Note that a subsequent pairing on this connection may change the
+ * encryption key, but not to a lower security level.
+ *
+ * The parameter @p authenticated is used to indicate whether the
+ * encryption key is considered authenticated by the GAP layer. If true,
+ * the connection will get BT_SECURITY_L4 security level, otherwise it
+ * will get BT_SECURITY_L2 security level. Access to Bluetooth services
+ * is controlled by this security level. The user of this API must
+ * consider the security implications of this.
+ *
+ * It is highly recommended that the shared secret is authenticated, and
+ * that unauthenticated secrets are not trusted.
+ *
+ * Requires @kconfig{CONFIG_NRF_BT_CONN_SET_LTK} to be enabled.
+ *
+ * @param conn Connection object.
+ * @param ltk Encryption key "LTK" to set.
+ * @param authenticated Whether GAP will consider the connection to be
+ * authenticated.
+ *
+ * @retval -EALREADY Too late. The encryption key is already set using
+ * this API, a bond exists, a pairing is in progress or has already been
+ * completed. Please unpair first.
+ * @retval -ENOMEM Could not allocate Host memory. Increase
+ * @kconfig{CONFIG_BT_MAX_PAIRED}.
+ * @retval -EINVAL Null pointer detected in arguments. This is undefined
+ * behavior when @kconfig{CONFIG_NO_RUNTIME_CHECKS} is enabled.
```

Comment:
I don't really have any experience with the CHECKIF function, but since this results in undefined behavior with CONFIG_NO_RUNTIME_CHECKS why don't we just have a regular if statement instead?